### PR TITLE
Get all favorites with playlist ls

### LINKED ISSUE #52: User can get all playlists with /playlists/:id/favorites endpoint
When the user sends a `GET api/v1/playlists/:id/favorites` request

User can expect a successful response body as follows with a `200 status`:
```
{
  "id": 1,
  "title": "Cleaning House",
  "songCount": 2,
  "songAvgRating": 27.5,
  "favorites" : [
                  {
                    "id": 1,
                    "title": "We Will Rock You",
                    "artistName": "Queen"
                    "genre": "Rock",
                    "rating": 25
                  },
                  {
                    "id": 4,
                    "title": "Back In Black",
                    "artistName": "AC/DC"
                    "genre": "Rock",
                    "rating": 30
                  }
               ],
    "createdAt": 2019-11-26T16:03:43+00:00,
    "updatedAt": 2019-11-26T16:03:43+00:00
}

```
Return `200 status` with an empty array if there are no playlist favorites in the database???

Acceptance criteria:
- [ ]  All tests passing
- [ ]  Happy path
- [ ]  Peer review

### PR DESCRIPTION
**Pull request by lrs8810**
## Type of change 

- [x] :hatching_chick: New feature (non-breaking change which adds functionality)
- [x] :page_with_curl: This change requires a documentation update

## Summary of changes 
- Add spec for happy path `GET /playlists/:id/favorites/:id`
- Add route for `GET /playlists/:id/favorites/:id`

## How Has This Been Tested?

- [x] :black_square_button: Integration testing 

## Checklist:

- [x] :white_check_mark: I have performed a self-review of my own code
- [x] :eight_spoked_asterisk: I have commented my code, particularly in hard-to-understand areas
- [x] :no_entry_sign: My changes generate no new warnings
- [x] :heavy_check_mark: I have added tests that prove my fix is effective or that my feature works
- [x] :100: New and existing unit tests pass locally with my changes

### CODE REVIEW

File: routes/api/v1/playlists.js [hunk 1]
```diff
@@ -84,10 +84,43 @@ router.delete('/:id', (request, response) => {
   });
 })
 
+router.get('/:playlistId/favorites', (req, res) => {
+  var playlistId = req.params.playlistId;
+  return database('playlists')
```
Comment (lrs8810) [inline routes/api/v1/playlists.js]: Would like to refactor during a later sprint.

File: tests/playlistFavorites.spec.js [hunk 2]
```diff
@@ -51,3 +51,74 @@ describe('Test it can add favorites to playlists_favorites endpoint', () => {
...
+      await database('favorites').insert(favorite_1)
+      await database('favorites').insert(favorite_2)
+      await database('favorites').insert(favorite_3)
+      await database('playlists_favorites').insert(playlistFavorite_1)
+      await database('playlists_favorites').insert(playlistFavorite_2)
+
+      const res = await request(app)
+        .get('/api/v1/playlists/9/favorites')
+
+      expect(res.statusCode).toBe(200);
```
Comment (lrs8810) [inline tests/playlistFavorites.spec.js]: Are there any other tests you think we should add?

File: routes/api/v1/playlists.js (hunk 1 above)
Comment (zacisaacson) [reply routes/api/v1/playlists.js]: Definitely could refactor all three database calls into helper methods at a later time

File: tests/playlistFavorites.spec.js [hunk 3]
```diff
@@ -51,3 +51,74 @@ describe('Test it can add favorites to playlists_favorites endpoint', () => {
...
+      expect(res.body).toHaveProperty("id");
+      expect(res.body).toHaveProperty("songCount");
+      expect(res.body.songCount).toBe(2);
+      expect(res.body).toHaveProperty("songAvgRating");
+      expect(res.body.songAvgRating).toBe(77.5);
+      expect(res.body).toHaveProperty("favorites");
+      expect(res.body.favorites.length).toBe(2);
+      expect(res.body).toHaveProperty("createdAt");
+      expect(res.body).toHaveProperty("updatedAt");
```
Comment (zacisaacson) [inline tests/playlistFavorites.spec.js]: Testing looks great!

**Pull request merged**